src/hooks/useNotificationTypes.tsx

```diff
@@ -1026,6 +1030,202 @@ export const notificationTypes: NotificationTypeConfig[] = [
+          setEnabled(false);
+          return;
+        }
+
+        const now = Date.now();
+        newDeposits.forEach((deposit) => {
+          if (
+            now - lastNotificationTimeRef.current > NOTIFICATION_DEBOUNCE &&
+            !prevDepositIdsRef.current.has(deposit.id)
+          ) {
+            trigger({
+              id: `deposit-${deposit.id}`,
+              displayData: {
+                toastDuration: DEFAULT_TOAST_AUTO_CLOSE_MS * 2,
+                groupKey: 'DepositAddressEvents',
+                slotTitleLeft: (
+                  <Icon iconName={IconName.Signal} size="1.25rem" tw="text-color-success" />
+                ),
+                title: stringGetter({ key: STRING_KEYS.DEPOSIT_DETECTED_TITLE }),
+                body: stringGetter({
+                  key: STRING_KEYS.DEPOSIT_DETECTED_BODY,
+                  params: {
+                    AMOUNT: Number(deposit.amount).toFixed(2),
+                    ASSET: 'USDC',
+                  },
+                }),
+                toastSensitivity: 'foreground',
+                actionDescription: stringGetter({ key: STRING_KEYS.VIEW_TRANSACTION }),
+                renderActionSlot: () => {
+                  const chainInfo = CHAIN_INFO[deposit.chain_id];
+                  return (
+                    chainInfo?.explorerBaseUrl && (
+                      <Link
+                        href={`${chainInfo.explorerBaseUrl}/tx/${deposit.transaction_hash}`}
+                        isAccent
+                      >
+                        {stringGetter({ key: STRING_KEYS.VIEW_TRANSACTION })}
+                      </Link>
+                    )
+                  );
+                },
+              },
+              updateKey: [deposit.id],
+            });
+            prevDepositIdsRef.current.add(deposit.id);
+          }
+          lastNotificationTimeRef.current = now;
+        });
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+      }, [newDeposits, trigger, stringGetter]);
+
+      // Stage 2: Watch for wallet balance increases (after sweep/rebalance to subaccount)
+      useEffect(() => {
+        if (!usdcAmount) return;
+
+        const now = Date.now();
+        const prevWallet = prevWalletBalanceRef.current;
+
+        if (!prevWallet) {
+          prevWalletBalanceRef.current = usdcAmount;
+          return;
+        }
+
+        const prevWalletBalanceBN = MaybeBigNumber(prevWallet);
+        const usdcBalanceBN = MaybeBigNumber(usdcAmount);
+
+        if (prevWalletBalanceBN && usdcBalanceBN && usdcBalanceBN.gt(prevWalletBalanceBN)) {
+          const diff = usdcBalanceBN.minus(prevWalletBalanceBN);
+          const matchingDeposit = newDeposits.find(
+            (deposit) => Number(deposit.amount).toFixed(2) === diff.toFixed(2)
+          );
+
+          if (now - lastNotificationTimeRef.current > NOTIFICATION_DEBOUNCE) {
+            trigger({
+              id: `deposit-confirmed-${matchingDeposit?.id}`,
```

Comment:
if `matchingDeposit` is undefined, we should use `crypto.randomUUID()` otherwise we would have duplicate `...-undefined` notification ids